src/assemble_workflow/bundle_rpm.py

```diff
@@ -89,7 +89,10 @@ def build(self, name: str, dest: str, archive_path: str, build_cls: BuildManifes
         os.environ.pop('OPENSEARCH_PATH_CONF', None)
 
         # Restore config file and core folder to original location
-        shutil.move(f"{min_bin_env_path}.backup", min_bin_env_path)
+        if os.path.exists(f"{min_bin_env_path}.backup"):
+            logging.info(f"Restore {min_bin_env_path}.backup to {min_bin_env_path}")
+            shutil.move(f"{min_bin_env_path}.backup", min_bin_env_path)
```

Comment:
This looks hacky, why is any of this necessary? Why is it undoing something done in extract?

Reply:
This was in for the opensearch only change.
The reason being there is a hardcoded `source /etc/sysconfig/opensearch` line specifically in rpm version of core.
This caused the failure during installing of plugin zips.

In extract block we comment out the line and back up the original file.
Here we just move the backup back.

The change for dashboards is just to check if the file exist, if not then ignore this step.

Thanks.

Reply:
Why isn't that line removed in the OpenSearch repo in the first place, why is this happening here?

I'll let it slide, but this is a hack on top of a hack.

Reply:
Improvement issue: https://github.com/opensearch-project/opensearch-build/issues/1864